### PR TITLE
[3006.x] fix: file.directory state children_only kwarg did not work

### PR DESCRIPTION
**Pull request by raddessi**
### What does this PR do?
This fixes the behavior of the `children_only` kwarg on the file.directory state to perform as documented.

I'm testing against 3006.1

### What issues does this PR fix or reference?
Fixes: n/a

### Previous Behavior
The root directory would still be modified even when `children_only` is set to True

### New Behavior
Now it is not modified when requested.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
  - Not needed since we are bringing functionality inline with current docs.
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by MKLeb** [comment]
@raddessi Looks great! I see you're planning on adding a changelog and tests, so once those are in we can review this :)

**Comment by raddessi** [comment]
I hope not having an issue is OK. I used the PR number as stated in the changelog docs.

**Comment by raddessi** [comment]
Hmm  I clearly do not understand how file mode syntax works with the new pytest suite utils.. any insight would be helpful.

**Review by whytewolf — DISMISSED** [review]

**Review by dmurphy18 — DISMISSED** [review]

**Comment by raddessi** [comment]
@MKLeb when you have time this has been sitting a while, can you let me  know if it needs changes?

**Review by MKLeb — DISMISSED** [review]

**Review by s0undt3ch — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by Ch3LL — DISMISSED** [review]

File: tests/pytests/functional/states/file/test_directory.py [hunk 1]
```diff
@@ -113,6 +113,49 @@ def _get_oct_mode(name):
             assert _mode == _get_oct_mode(untouched_dir)
 
 
+def test_directory_children_only(file, tmp_path):
+    """
+    file.directory with children_only=True
+    """
+    if IS_WINDOWS:
+        pytest.skip("Skipped on windows")
```
Comment (s0undt3ch) [inline tests/pytests/functional/states/file/test_directory.py:121]:
```suggestion
@pytest.mark.skip_on_windows
def test_directory_children_only(file, tmp_path):
    """
    file.directory with children_only=True
    """
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by raddessi** [comment]
Test failures seem to be unrelated

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**